markers/env.py

```diff
@@ -0,0 +1,27 @@
+from utils import testgen
+#def pytest_collection_modifyitems(session, config, items):
+#    import pdb; pdb.set_trace()
+
+BROWSERS = ['firefox', 'chrome', 'ie']
+TCPSTACKS = ['ipv4', 'ipv6']
+
+
+def process_env_mark(metafunc, mark_name, choices):
+    if hasattr(metafunc.function, mark_name):
+        if getattr(metafunc.function, mark_name).args:
+            browser_mark = getattr(metafunc.function, mark_name).args[0]
+        else:
+            raise Exception('No keyword given to mark')
+        if browser_mark == 'all':
+            metafunc.fixturenames.append(mark_name)
+            testgen.parametrize(metafunc, mark_name, choices)
+        elif browser_mark == 'none':
+            return
```

Comment:
it seems there should be handling of explicit listing of the choices per parameterization